src/analyze/analyze.c

```diff
@@ -1551,6 +1553,27 @@ static int dump_syscall_filters(int argc, char *argv[], void *userdata) {
 }
 #endif
 
+static int dump_timespan(int argc, char *argv[], void *userdata) {
+        int ret = 0, r;
+        char **input_timespan = argv + 1;
+        usec_t default_unit = 1 /* usec */, output_usecs;
+
+        if (argc == 3) {
+                char **multiplier = argv + 2;
+                extract_multiplier(*multiplier, &default_unit);
```

Comment:
this would be a bit sloppy, because specifiyng "sec foobarxxxxyzzzz" would be parsed cleanly, even though it really shouldn't...

Reply:
For this one, do you have a suggestion if it's to be kept this way? I notice that `extract_multiplier` currently has no way to return match/no match status because currently in the way it interacts with `parse_time` it doesn't have to, but in this case to avoid this I think we have to know whether we actually matched something by returning `int` and changing the return to happen as a mutable argument, perhaps?

Reply:
(Also, I was somewhat surprised that `extract_multiplier` checks "starts with" instead of doing direct equality matching.)

Reply:
well, extract_multipler() is an internal call so far, that is used exclusively while we iteratively parse a time specification. i.e. when we parse a string such as "5s 153ms 5us" we use it to parse the "s", the "ms" and the "us", and each time we need to continue parsing right after, hence it works the way it works, that it uses startswith() and returns where we left of.

That said, I am still not convinced that specifying the default unit is really useful. Where's the benefit of being able to write "5" "s" in two arguments rather than requiring "5s" in one?